### PR TITLE
[6068] When an user updates the page-url the sidebar should refresh his data

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/6068

### CODE REVIEW

File: ui/app/src/state/epics/pathNavigator.ts [hunk 1]
```diff
@@ -399,7 +399,10 @@ export default [
             actions.push(pathNavigatorRefresh({ id: navigator.id }));
           } else if (!navigator.isRootPathMissing && navigator.currentPath.startsWith(sourcePath)) {
             actions.push(pathNavigatorSetCurrentPath({ id: navigator.id, path: navigator.rootPath }));
-          } else if (navigator.currentPath === parentOfTargetPath || navigator.currentPath === parentOfSourcePath) {
+          } else if (
+            withoutIndex(navigator.currentPath) === parentOfTargetPath ||
+            withoutIndex(navigator.currentPath) === parentOfSourcePath
```
Comment (rart) [inline ui/app/src/state/epics/pathNavigator.ts:404]: What's the logic?

Would you need to equalise the `parentOf*` to also strip index for this comparison. Make sure you're not looking at the issue through a straw.

File: ui/app/src/state/epics/pathNavigatorTree.ts [hunk 2]
```diff
@@ -338,14 +338,18 @@ export default [
             [parentPathOfTargetPath, parentPathOfSourcePath].forEach((path) => {
               if (
                 // If in totalByPath is an item that has been loaded and must update...
-                path in tree.totalByPath
+                // path in totalByPath may be a page, and its path has index.xml, so it needs to be validated too.
+                path in tree.totalByPath ||
+                withIndex(path) in tree.totalByPath
```
Comment (rart) [inline ui/app/src/state/epics/pathNavigatorTree.ts:343]: Does it make sense to assure to comparison is made both ways?
```suggestion
                withoutIndex(path) in tree.totalByPath ||
                withIndex(path) in tree.totalByPath
```

File: ui/app/src/state/epics/pathNavigatorTree.ts [hunk 3]
```diff
@@ -338,14 +338,18 @@ export default [
...
+                // Get correct path to fetch (may include index.xml)
+                const fetchPath = path in tree.totalByPath ? path : withIndex(path);
                 // If its children are loaded, then re-fetch to get the new
-                tree.childrenByParentPath[path] &&
+                tree.childrenByParentPath[fetchPath] &&
                   actions.push(
                     pathNavigatorTreeFetchPathChildren({
                       id: id,
-                      path: path,
+                      path: fetchPath,
```
Comment (rart) [inline ui/app/src/state/epics/pathNavigatorTree.ts:352]: Please explain changes

**Review by rart — COMMENTED** [review]
Please explain what the issue is/was and how these changes address.

File: ui/app/src/state/epics/pathNavigator.ts (hunk 1 above)
Comment (jvega190) [reply ui/app/src/state/epics/pathNavigator.ts:404]: `parentOf*` is retrieved by using `getParentPath` that already uses `withoutIndex`. A refresh of the pathNavigator is done when the currentPath is the same as the parent of the target or source. The issue was that the current path was matching one of the parents, but not entiretly due to the /index.xml being present in the currentPath

File: ui/app/src/state/epics/pathNavigatorTree.ts (hunk 2 above)
Comment (jvega190) [reply ui/app/src/state/epics/pathNavigatorTree.ts:343]: `path` is one of the parents (parentPathofTargetPath or parentPathOfSourcePath), and they're retrieved using `getParentPath`. This is the same scenario as the one above, where `getParentPath` already removes the index.xml from the path

**Review by rart — APPROVED** [review]

**Pull request merged**